### DISCUSSION TITLE
The "--with" option does not exist

### DISCUSSION THREAD

**Discussion by ghost** [python-poetry/poetry#8686] [General]
Hi, I am installing Poerty in Ubuntu 22.04 (Virtual Box 7) version and it is displaying an error: 
NoSuchOptionException
The "'--with" option does not exist.

Anyone having the same issue? Did you fixed it?

**Comment by finswimmer** [comment]
What version of Poetry are you using? What exact command have you tried?

**Comment by ghost** [comment]
Version 1.7.0.
In the following video (4:52) he runs in a Mac and it works:
[https://www.youtube.com/watch?v=XFiof0V3nhA&list=PLeiGn0LnIdiFLljbpx50pYq-fjFo9b28T&index=6](url)
The command is: 
`poetry install --with ui,local`

In other tutorials (also using a Mac) they run the following command, and it works: 
`poetry install --with ui`

Please any help.

**Comment by dimbleby** [comment]
well that option does exist, so I guess you are not using version 1.7.0

**Comment by ghost** [comment]
See this image, it confirms 1.7.0 version is installed on my system:
![image](https://github.com/python-poetry/poetry/assets/49165589/98ece71f-0920-470d-a992-ef3fae6fd55e)

**Reply by dimbleby** [reply]
nevertheless that image confirms that you are not using poetry 1.7.0, clikit has not been part of poetry for several years now.

you've messed up your poetry installation somehow: uninstall all the poetry that you can find and start over

**Comment by ghost** [comment]
This is the cmmand i used to install:
`brew install poetry`

Just found this articule on why it is not recommended to instal poetry from brew:
[(https://stackoverflow.com/questions/73365780/why-is-not-recommended-to-install-poetry-with-homebrew)]

**Reply by abn** [reply]
I'm confused. You say you use brew, but your screenshot is of Ubuntu and executing inside a VirtualBox? Does brew run on Linux distros? 

In any case, nuke your poetry installation and either use the install script or install using pipx.

**Reply by abn** [reply]
https://python-poetry.org/docs/#installation

**Comment by Menen21** [comment]
Got the same problem, I guess you are using WSL, just restart your terminal and it works fine.

Hope it solves your issue.

**Reply by alexiswl** [reply]
> Got the same problem, I guess you are using WSL, just restart your terminal and it works fine.

If I install poetry in a virutal env but I have poetry in my standard PATH i.e `/usr/bin/poetry`, 

when I run `type poetry` I get

```
poetry is hashed (/usr/bin/poetry)
```

If I re-source my environment after local virtual env I get the right version of poetry.  

Likely what is happening with your WSL2 above
